### ISSUE TITLE
[FEATURE] TimeGapSplit add n_splits param

### ISSUE THREAD

**Issue by stephanecollot** [koaning/scikit-lego#232]
Currently in TimeGapSplit you can specify exactly the behaviour in production with:
- `train_duration`
- `valid_duration`
- `gap_duration`

and the class will define for you automatically the number folds.

We could add a new parameter `n_splits`, that will allow to give for example only `gap_duration` and `valid_duration`. The use case is the following: usually we know the gap duration (given by target definition) and validation duration (given by the retraining frequency), and we usually want 5 folds for hyper-parameters tuning.

In other words, we will have 4 parameters, and you need to provide 3 of them.
